pre_commit/languages/docker.py

```diff
@@ -74,15 +75,18 @@ def install_environment(
 
 
 def docker_cmd():  # pragma: windows no cover
+    superproject_dir = git.get_superproject_root()
+    project_dir = git.get_root()
+    project_relpath = os.path.relpath(project_dir, superproject_dir)
     return (
         'docker', 'run',
         '--rm',
         '-u', '{}:{}'.format(os.getuid(), os.getgid()),
         # https://docs.docker.com/engine/reference/commandline/run/#mount-volumes-from-container-volumes-from
         # The `Z` option tells Docker to label the content with a private
         # unshared label. Only the current container can use a private volume.
-        '-v', '{}:/src:rw,Z'.format(os.getcwd()),
-        '--workdir', '/src',
+        '-v', '{}:/src:rw,Z'.format(superproject_dir),
+        '--workdir', os.path.abspath(os.path.join('/src', project_relpath)),
```

Comment:
this isn't going to work if the submodule is nested more than one level into the parent project